packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -213,6 +213,8 @@ export function getBeaconBlockApi({
 
       // TODO: Validate block
 
+      chain.blockProposerIndices.set(toHexString(signedBlock.message.stateRoot), signedBlock.message.proposerIndex);
```

Comment:
You will want to track blocks by block root, not state root.
You will need to use the fork choice to get all ancestors of the recently finalized block and iterate thru that list to check against this map.
You will need to ensure that you check the fork choice before it is pruned, since it is pruned on finalization.